shared_model/interfaces/base/model_primitive.hpp

```diff
@@ -29,7 +30,7 @@ namespace shared_model {
      * @tparam Model - your new style model
      */
     template <typename Model>
-    class ModelPrimitive {
+    class ModelPrimitive : public Cloneable<ModelPrimitive<Model>> {
```

Comment:
What is the point in moving away cloning function from ModelPrimitive?

Reply:
To use it somewhere else besides shared_model

Reply:
Not sure that it is needed